src/aics-image-viewer/components/ViewerStateProvider/index.tsx

```diff
@@ -171,20 +185,26 @@ const ViewerStateProvider: React.FC<{ viewerSettings?: Partial<ViewerState> }> =
 
   const context = useMemo(() => {
     ref.current = {
+      ...ref.current,
       ...viewerSettings,
       channelSettings,
       changeViewerSetting,
       setChannelSettings,
       changeChannelSetting,
       applyColorPresets,
+      // Reset-related callbacks will be set by the ResetStateProvider
     };
 
     // `ref` is wrapped in another object to ensure that the context updates when state does.
     // (`ref` on its own would always compare equal to itself and the context would never update.)
     return { ref };
   }, [viewerSettings, channelSettings]);
 
-  return <ViewerStateContext.Provider value={context}>{props.children}</ViewerStateContext.Provider>;
+  return (
+    <ViewerStateContext.Provider value={context}>
+      <ResetStateProvider viewerStateInputProps={props.viewerSettings}>{props.children}</ResetStateProvider>
```

Comment:
what is in props.children?

Reply:
It's passed in from the `ViewerStateProvider`, so typically this would be the viewer component and the rest of the app. Basically anything `ViewerStateProvider` is wrapping

Reply:
Marking as resolved now that ResetStateProvider is a class